simulariumio/cellpack/cellpack_converter.py

```diff
@@ -0,0 +1,328 @@
+        all_ingredients,
+        time_step_index: int,
+        scale_factor: float,
+        box_center: np.array,
+        geo_type: DISPLAY_TYPE,
+    ) -> AgentData:
+        dimensions = CellpackConverter._parse_dimensions(all_ingredients)
+        spatial_data = AgentData.from_dimensions(dimensions)
+        display_data = {}
+        agent_id_counter = 0
+        for ingredient in all_ingredients:
+            ingredient_data = ingredient["recipe_data"]
+            ingredient_key = ingredient_data["name"]
+            ingredient_results_data = ingredient["results"]
+            agent_display_data = CellpackConverter._get_ingredient_display_data(
+                geo_type, ingredient_data
+            )
+            display_data[ingredient_key] = DisplayData(
+                name=ingredient_key,
+                display_type=agent_display_data["display_type"],
+                url=agent_display_data["url"],
+            )
+            if len(ingredient_results_data["results"]) > 0:
+                for j in range(len(ingredient_results_data["results"])):
+                    CellpackConverter._unpack_positions(
+                        ingredient_results_data,
+                        time_step_index,
+                        ingredient_key,
+                        j,
+                        agent_id_counter,
+                        spatial_data,
+                        scale_factor,
+                        box_center,
+                    )
+                    agent_id_counter += 1
+            elif ingredient_results_data["nbCurve"] > 0:
+                for i in range(ingredient_results_data["nbCurve"]):
+                    CellpackConverter._unpack_curve(
+                        ingredient_results_data,
+                        time_step_index,
+                        ingredient_key,
+                        i,
+                        agent_id_counter,
+                        spatial_data,
+                        scale_factor,
+                        box_center,
+                    )
+                    agent_id_counter += 1
+        spatial_data.display_data = display_data
+        return spatial_data
+
+    @staticmethod
+    def _update_meta_data(meta_data: MetaData, box_size: np.array) -> MetaData:
+        camera_z_position = box_size[2] if box_size[2] > 10 else 100.0
+        meta_data.camera_defaults = CameraData(
+            position=np.array([10.0, 0.0, camera_z_position]),
+            look_at_position=np.array([10.0, 0.0, 0.0]),
+            fov_degrees=60.0,
+        )
+
+    @staticmethod
+    def _read(input_data: CellpackData) -> TrajectoryData:
+        """
+        Return a TrajectoryData object containing the Cellpack data
+        """
+        print("Reading Cellpack Data -------------")
+        # currently only converts one model, ie one time step
+        time_step_index = 0
+        # default scale for simularium => cellpack
+        input_data.meta_data.scale_factor *= 0.1
```

Comment:
Just double checking this is supposed to be `*=` and not `=`?

Reply:
yes, but I could put a better comment here. By default going from cellpack, it needs to be scaled down by 0.1. But we also have this user entered scale. I need to add this to the documentation too, but we figured instead of a user override they could do the scale number they relative to cellpack, and then it would adjust for simularium